torch/_inductor/kernel/conv.py

```diff
@@ -357,11 +376,12 @@ def channels_last_conv():
     # TODO: check if it's beneficial to convert Conv1d to Conv2d and then
     # apply channels last.
     if V.graph.layout_opt and ndim == 2:
-        V.graph.num_channels_last_conv += 1
-        x = ir.ExternKernel.require_channels_last(x)
-        # TODO maybe we can convert weights to channels last just once before
-        # running the model.
-        weight = ir.ExternKernel.require_channels_last(weight)
+        if inductor_config.conv_force_channels_last:
```

Comment:
this is making it so that we are no longer forcing channels last when `inductor_config.conv_force_channels_last` is False but `layout_opt` is true.

Also we should remove the inline logic for `conv_force_channels_last` and just take it into account in `decide_layout_opt`

Reply:
@pragupta was able to take the comment and submit a PR which is much cleaner using layout_opt could you take a look at https://github.com/pytorch/pytorch/pull/107812

This PR will now focus on adding the device_type into compile_meta and getting our ROCm compatible triton configs